File: src/main/java/com/widehouse/whatnext/domain/TaskRepository.java
package com.widehouse.whatnext.domain;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

public interface TaskRepository extends JpaRepository<Task, Integer> {
    List<Task> findByStatus(TaskStatus status);

    List<Task> findByCategory(Category category);
}

File: src/main/kotlin/com/widehouse/whatnext/domain/TaskRepository.kt
package com.widehouse.whatnext.domain

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor

interface TaskRepository : JpaRepository<Task, Int>, JpaSpecificationExecutor<Task> {
    fun findByStatus(status: TaskStatus): List<Task>

    fun findByCategory(category: Category): List<Task>

    fun findByCategoryAndStatus(category: Category, status: TaskStatus): List<Task>
}